src/archive/zip.rs

```diff
@@ -28,16 +29,29 @@ use crate::{
 
 /// Unpacks the archive given by `archive` into the folder given by `output_folder`.
 /// Assumes that output_folder is empty
-pub fn unpack_archive<R>(mut archive: ZipArchive<R>, output_folder: &Path, quiet: bool) -> crate::Result<usize>
+pub fn unpack_archive<R>(
+    mut archive: ZipArchive<R>,
+    output_folder: &Path,
+    password: Option<impl AsRef<[u8]>>,
+    quiet: bool,
+) -> crate::Result<usize>
 where
     R: Read + Seek,
 {
     assert!(output_folder.read_dir().expect("dir exists").count() == 0);
 
     let mut unpacked_files = 0;
 
+    let password = password.as_ref().map(|p| p.as_ref().to_owned());
+
     for idx in 0..archive.len() {
-        let mut file = archive.by_index(idx)?;
+        let mut file = match password.clone() {
+            Some(password) => archive
+                .by_index_decrypt(idx, password.to_owned().as_bytes())
+                .unwrap()
+                .map_err(|_| zip::result::ZipError::UnsupportedArchive("Password required to decrypt file"))?,
```

Comment:
You can replace these 2 lines by the double try operator `??`.

No need to map the error because `zip` will give the detailed information of why this failed, and our error treatment is set in a way that gives a good error message.